src/inmanta/ast/type.py

```diff
@@ -161,6 +213,51 @@ def __eq__(self, other: object) -> bool:
             return NotImplemented
         return self.element_type == other.element_type
 
+    def is_primitive(self) -> bool:
+        return self.element_type.is_primitive()
+
+    def corresponds_to(self, type: Type) -> bool:
+        if isinstance(type, Any):
+            return True
+        if not isinstance(type, NullableType):
+            return False
+        return self.element_type.corresponds_to(type.element_type)
+
+    def as_python_type_string(self) -> "str | None":
+        return f"{self.element_type.as_python_type_string()} | None"
+
+    def to_python(self, instance: object) -> "object":
+        if isinstance(instance, NoneValue):
+            return None
+        return self.element_type.to_python(instance)
+
+    def has_custom_to_python(self) -> bool:
+        return self.element_type.has_custom_to_python()
+
+
+class Any(Type):
+    """
+    this type represents the any type, similar to Python's
+
+    the Any class itself is neither the top nor the bottom type in the type hierarchy.
+    """
+
+    def corresponds_to(self, type: Type) -> bool:
+        return True
+
+    def as_python_type_string(self) -> "str | None":
+        return "object"
+
+    def has_custom_to_python(self) -> bool:
+        return False
```

Comment:
Do we need all these explicit `return False` implementations (here, `Primitive`, ...)? The `Type` implementation should cover that.

Reply:
I wonder if type shouldn't be an ABC (well, that ship has sailed).

Its implementation of has_custom_to_python makes no sense in general.

Reply:
I cleaned it up some more

Reply:
> that ship has sailed

Many ships, when it comes to this module.